distro/commands/distro.go

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 The original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package commands
+
+import (
+	"github.com/projectriff/riff/cmd/commands"
+	"github.com/projectriff/riff/pkg/core"
+	"github.com/spf13/cobra"
+)
+
+func DistroCreate(c *core.ImageClient) *cobra.Command {
+	options := core.CreateDistroOptions{}
+
+	command := &cobra.Command{
+		Use:     "create",
+		Short:   "Create a riff distribution.",
+		Long:    "Create a riff distribution (a zipped tar archive or .tgz file) from a given riff manifest.",
+		Example: `  riff-distro create --output=./riff.tgz`,
+		Args:    cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := (*c).CreateDistro(options)
+			if err != nil {
+				return err
+			}
+
+			commands.PrintSuccessfulCompletion(cmd)
+			return nil
+		},
+	}
+
+	command.Flags().StringVarP(&options.Manifest, "manifest", "m", "stable", "manifest for the download; can be a named manifest (stable or latest) or a path of a manifest file")
```

Comment:
I'll take the less easy path but https://github.com/projectriff/riff/issues/929 needs fixing first.

Reply:
The less easy path turned out to be rather tricky because of the relative timing of injection of configuration and when command help runs. Also, the problem is more widespread than just in this command. For instance, the same problem occurs in `riff system install`.

Raised https://github.com/projectriff/riff/issues/933 to address consistently.